src/segment_language.go

```diff
@@ -1,18 +1,40 @@
 package main
 
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
 type loadContext func()
 
 type inContext func() bool
 
+type version struct {
+	full         string
+	major        string
+	minor        string
+	patch        string
+	versionRegex string
+	urlTemplate  string
+}
+
+func (v *version) string() string {
+	version, err := TruncatingSprintf(v.urlTemplate, v.full, v.major, v.minor, v.patch)
```

Comment:
The url creation should not be in here, all it should do is return the version, everything else should be handled in `language.go`.